src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -544,6 +544,42 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
             })
     }
 
+    /// Resolve a type alias for the purposes of intra-doc links.
+    fn resolve_type_alias(&self, ty_id: DefId) -> Option<Res> {
```

Comment:
I don't think this is a good description; this is the *use-case*, but not what it's doing.
```suggestion
    /// Convert a DefId to a Res, where possible.
    ///
    /// This is used for resolving type aliases.
    fn def_id_to_res(&self, ty_id: DefId) -> Option<Res> {
```